src/components/ErrorBoundary/withErrorBoundary.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * @flow
+ * @file withErrorBoundary HOC
+ * @author Box
+ */
+
+import * as React from 'react';
+import ErrorBoundary from './ErrorBoundary';
+
+type Props = {
+    onComponentError?: Function,
+};
+
+const withErrorBoundary = (WrappedComponent: React.ComponentType<any>) => {
+    // $FlowFixMe doesn't know about forwardRef (https://github.com/facebook/flow/issues/6103)
+    return React.forwardRef(
+        ({ onComponentError, ...rest }: Props, ref: React.Ref<any>) => (
```

Comment:
That was the idea. I was hesitant to tie into the existing onError because I'm not sure where/how it's being used or what's being passed to it. I'll investigate a bit more and update if appropriate.

Reply:
ContentPreview in particular expects a very specific type of error (PreviewError from box-content-preview). I don't think mixing them without further thought would be wise. I'll just remove the handler for now and we can address it later if users request it.